src/views/pages/Inbox/index.js

```diff
@@ -31,28 +31,25 @@ const test_messages = [
   }
 ];
 
-function Inbox(props) {
+function Inbox() {
   const {messageService} = useServiceContext();
   const [messageList, setMessageList] = useState(null);
+  const defaultIdentity = useDefaultIdentity();
 
   useEffect(() => {
     const sub = messageService.received_messages$.subscribe((d) => {
       setMessageList(d && d.length > 0 ? d : null);
     });
 
     let id = setInterval(() => {
-      messageService.checkMessages(props.user_identity);
+      messageService.checkMessages(defaultIdentity);
     }, 5000);
 
     return () => {
       sub.remove();
       clearInterval(id);
     };
-  }, []);
-
-  useEffect(() => {
-    messageService.checkMessages(props.user_identity);
-  }, [props.user_identity]);
+  }, [defaultIdentity, messageService]);
```

Comment:
- the `messageService` will never change
- sometimes you want to create multiple `useEffect` hooks because you want some functions to run once and others to re-run when some state changes -- in this case you want lines 40-42 in its own separate useEffect with an empty dependency array and lines 44-46 in another useEffect hook with just the `defaultIdentity` in the dependency array. The respective cleanup logic would also be separated then.

Reply:
Yeah, that messageService bit there was because my linter kept complaining. I saw we had that separated into two separate useEffect hooks but to be honest, neither of those dependencies should ever really change.

Reply:
In the rare, rare case that either of these change, cool! We just updated. Otherwise, this doesn't really need to run more than once but there's no reason I can see that it needs to _only_ run once.